tensorflow/core/common_runtime/next_pluggable_device/next_pluggable_device.cc

```diff
@@ -126,6 +128,77 @@ Status NextPluggableDevice::MakeTensorFromProto(
   Status status;
   if (alloc_attrs.on_host()) {
     *tensor = parsed;
+  } else if (parsed.dtype() == DT_VARIANT) {
+    const Variant* from = parsed.flat<Variant>().data();
+    Tensor copy(cpu_allocator(), DT_VARIANT, parsed.shape());
+    Variant* copy_variant = copy.flat<Variant>().data();
+
+    std::list<Notification> notifications;
+    Status copy_status;
+    auto copier = [this, &alloc_attrs, &notifications, &copy_status](
+                      const Tensor& from, Tensor* to) {
+      // Copier isn't run in a multithreaded environment, so we don't
+      // have to worry about the notifications list being modified in parallel.
+      notifications.emplace_back();
+      Notification& n = *notifications.rbegin();
+
+      StatusCallback done = [&n, &copy_status](const Status& s) {
+        if (copy_status.ok()) {
+          copy_status.Update(s);
+        }
+        n.Notify();
+      };
+      if (!DMAHelper::CanUseDMA(&from)) {
+        Status err = errors::Internal("NextPluggableDevice copy from non-DMA ",
```

Comment:
nit: right now abls::Status is preferred. Do you mind update the tsl errors to absl errors? Thanks!

Reply:
done

Reply:
@jyingl3 do you have any update on this PR? thanks

Reply:
Sorry I did not get notification and missed the update. Does the BUILD need to be updated? Otherwise it looks good.